### PR TITLE
SelfSubjectAccessReview test coverage

### PR DESCRIPTION
**Pull request by Skyterix1991**

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    @Test
+    @Order(2)
+    public void testLoadFromFile() throws IOException {
+        KubernetesClient client = server.getClient();
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+
+        SelfSubjectAccessReview review = mapper.readValue(getClass().getResourceAsStream("/test-selfsubjectaccessreview.yml"), SelfSubjectAccessReview.class);
```
Comment (manusa) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java]: Please use Serialization.unmarshal

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@EnableRuleMigrationSupport
+public class SelfSubjectAccessReviewAuthTest {
+
+    @Rule
+    public KubernetesServer server = new KubernetesServer();
+
+    @Test
+    @Order(1)
+    public void testList() {
```
Comment (manusa) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java]: The content of the tests doesn't match what the method declares to be testing (only one I found to be consistent was `testLoadFromFile`)

In this case the test declares to be testing "list", but it's creating a new SelfSubjectAccessReview

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@EnableRuleMigrationSupport
+public class SelfSubjectAccessReviewAuthTest {
+
+    @Rule
+    public KubernetesServer server = new KubernetesServer();
+
+    @Test
+    @Order(1)
```
Comment (manusa) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java]: These methods are invoked against a mock server, so they can probably run in parallel, right?

If so, remove the `@Order` and `@TestMethodOrder` annotations.

**Pull request closed without merging**